tango_simlib/tests/test_tango_sim_generator.py

```diff
@@ -18,63 +18,131 @@
 
 from tango_simlib import tango_sim_generator
 from tango_simlib.tests import test_sim_test_interface
-from tango_simlib.utilities import helper_module, sim_xmi_parser, fandango_json_parser
+from tango_simlib.utilities import (helper_module, sim_xmi_parser,
+                                    fandango_json_parser, simdd_json_parser)
 from tango_simlib.utilities.testutils import ClassCleanupUnittestMixin
 
 MODULE_LOGGER = logging.getLogger(__name__)
 
+class BaseTest:
 
-class test_TangoSimGenDeviceIntegration(ClassCleanupUnittestMixin, unittest.TestCase):
+    class TangoSimGenDeviceIntegration(ClassCleanupUnittestMixin, unittest.TestCase):
```

Comment:
+1

Reply:
The base class contains functions that are named with `test` prefixing the function names. Due to this, Nosetest first calls the base class which has none type class variables and thus, makes the tests fail. 
From the suggestion in [this link from Stackoverflow](https://stackoverflow.com/questions/1323455/python-unit-test-with-base-and-sub-class), wrapping a class around the base class (in this case, `TangoSimGenDeviceIntegration`) would make Nosetest begin execution from the derived classes instead of the base class.